--- a/cosign/private/sign.bzl
+++ b/cosign/private/sign.bzl
@@ -61,7 +61,7 @@ def _cosign_sign_impl(ctx):
 
     fixed_args = []
     if ctx.attr.repository:
-        fixed_args.extend(["--repository", ctx.attr.repository])
+        fixed_args.extend(["--repository", ctx.expand_make_variables("repository", ctx.attr.repository, {})])
 
     ctx.actions.expand_template(
         template = ctx.file._sign_sh_tpl,
