--- a/cosign/private/sign.bzl
+++ b/cosign/private/sign.bzl
@@ -58,7 +58,10 @@ def _cosign_sign_impl(ctx):
             "{{cosign_path}}": cosign.cosign_info.binary.short_path,
             "{{jq_path}}": jq.jqinfo.bin.short_path,
             "{{image_dir}}": ctx.file.image.short_path,
-            "{{fixed_args}}": " ".join(["--repository", ctx.attr.repository]),
+            "{{fixed_args}}": " ".join([
+                "--repository",
+                ctx.expand_make_variables("repository", ctx.attr.repository, {}),
+            ]),
         },
     )
 
